### PR TITLE
Support for Gconsale and Gconsump 

### PR DESCRIPTION
**Pull request by totto82**
This is PR #2185 minus the solve every iteration change.

### CODE REVIEW

**Comment by atgeirr** [comment]
jenkins build this please

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
The failure is expected since this PR contains a VREP fix that improves UDQ_WCONPROD.

**Comment by totto82** [comment]
jenkins build this update_data please

**Comment by totto82** [comment]
jenkins build this opm-tests=207 please

File: opm/simulators/wells/BlackoilWellModel_impl.hpp [hunk 1]
```diff
@@ -1711,8 +1718,41 @@ namespace Opm {
                 //OPM_THROW(std::runtime_error, "Group " + group.name() + "FLD control for injecting groups not implemented" );
             }
 
-        } else if (group.isProductionGroup())
-        {
+            // Handle GCONSALE
+            if (schedule().gConSale(reportStepIdx).has(group.name())) {
```
Comment (atgeirr) [inline opm/simulators/wells/BlackoilWellModel_impl.hpp]: Just to clarify: the code here looks like it has no effect other than throwing if it detects an illegal or unhandled situation. Is that correct?

Comment (totto82) [reply opm/simulators/wells/BlackoilWellModel_impl.hpp]: Yes. It is on my todo list.

**Comment by totto82** [comment]
jenkins build this update_data please

**Comment by totto82** [comment]
jenkins build this opm-tests=207 please

**Comment by totto82** [comment]
Why are this failing?

**Comment by atgeirr** [comment]
> Why is this failing?

Multiple data-updating PRs in flight always cause problems. It was eventually resolved for the WSEGSICD PR, I'll rerun Jenkins here to see if it's ok now (at least the opm-tests PR only contains the single expected changed case, so I have hope).

**Comment by atgeirr** [comment]
jenkins build this opm-tests=207 please

**Pull request merged**